### DISCUSSION TITLE
Open desktop app from browser

### DISCUSSION THREAD

**Discussion by ctkai01** [wailsapp/wails#2617] [Q&A]
I want to open the application desktop from the browser, how to do it with a custom protocol like setAsDefaultProtocolClient in ElectronJS

**Comment by airtonix** [comment]
Without diving into how electron might have abstracted the actual process from you, it generally goes like this (I'll use linux and gnome-shell or kde as the example) : 

You want to register: `org-protocol://`

1. create a `filename-doesnt-matter.desktop` file in `~/.local/share/applications/filename-doesnt-matter.desktop` with something like https://superuser.com/a/309343/59080 or https://unix.stackexchange.com/a/497147/49317. (for info on the [desktop spec](https://developer-old.gnome.org/desktop-entry-spec/#mime-types) )
2. register `x-scheme-handler/org-protocol`  with `xdg-mime default filename-doesnt-matter.desktop x-scheme-handler/org-protocol`


Not sure what the exact process looks like for microsoft or apple

**Reply by AlbinoDrought** [reply]
If using an NSIS installer on Windows, can do this: https://github.com/electron-userland/electron-builder/issues/623#issuecomment-242521219

**Reply by airtonix** [reply]
And i think the process on macosx might involve plist files. not sure.
